tests/unit/core/sagas/testLanguageTools.js

```diff
@@ -0,0 +1,82 @@
+import SagaTester from 'redux-saga-tester';
+
+import * as api from 'core/api/languageTools';
+import { CLEAR_ERROR } from 'core/constants';
+import addonsReducer, {
+  fetchLanguageTools,
+  loadAddonResults,
+} from 'core/reducers/addons';
+import apiReducer from 'core/reducers/api';
+import languageToolsSaga from 'core/sagas/languageTools';
+import { dispatchClientMetadata, fakeAddon } from 'tests/unit/amo/helpers';
+import {
+  createFakeLanguageAddon,
+  createStubErrorHandler,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  let errorHandler;
+  let mockApi;
+  let sagaTester;
+
+  beforeEach(() => {
+    errorHandler = createStubErrorHandler();
+    mockApi = sinon.mock(api);
+    sagaTester = new SagaTester({
+      initialState: dispatchClientMetadata().state,
+      reducers: {
+        api: apiReducer,
+        addons: addonsReducer,
+      },
+    });
+    sagaTester.start(languageToolsSaga);
+  });
+
+  it('calls the API for language tools', async () => {
+    const addon = { ...fakeAddon, slug: 'fancy' };
+    const response = { results: createFakeLanguageAddon({ addon }) };
+
+    mockApi
+      .expects('languageTools')
+      .once()
```

Comment:
it's worth doing `withArgs({ api: apiState })` here

Reply:
True! Confusingly the `callApi` takes the `state` param but it really is `apiState`. I'll add the `withArgs` here but I've also filed #3346.